ddl/ddl_api.go

```diff
@@ -2035,6 +2035,12 @@ func (d *ddl) CreateIndex(ctx sessionctx.Context, ti ast.Ident, unique bool, ind
 		return errors.Trace(err)
 	}
 
+	// Check before put the job is put to the queue.
+	_, err = buildIndexColumns(t.Meta().Columns, idxColNames)
```

Comment:
Please add more comments, explain why we do a check here.

Reply:
Done